#!/usr/bin/env python
PACKAGE = "robot_vision"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("voxelLeafSize", double_t,0, 
	"set leaf size for Voxel grid filter", 0.01,  0, 0.1)

gen.add("removePlaneIterations", int_t, 0, 
	"max iterations",    100, 0,   500)

gen.add("distanceThreshold",    double_t,    0,
	"distance Threshold",  0.02, 0, 0.1)

gen.add("PercentageCloud",   double_t,   0, 
	"percentage Point Cloud",  40.0,0.0,100.0)


gen.add("tolerance",   double_t,   0, 
	"Cluster tolerance",  0.03,0.0,0.1)

gen.add("minClustersSize",   int_t,   0, 
	"min Cluster Size",  100,10,50000)

gen.add("maxClustersSize",   int_t,   0, 
	"max Cluster Size",  25000,10,50000)


size_enum = gen.enum([ gen.const("Small",      int_t, 0, "A small constant"),
                       gen.const("Medium",     int_t, 1, "A medium constant"),
                       gen.const("Large",      int_t, 2, "A large constant"),
                       gen.const("ExtraLarge", int_t, 3, "An extra large constant")],
                     "An enum to set size")

gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "robot_vision", "robot_vision_params"))